File: app/models/api/entry.rb
module Api
  class Entry
    def initialize(entry:)
      @entry = entry
    end

    attr_reader :entry

    def attributes
      {
        id: entry.id,
        description: entry.description,
        eyeCatchingImage: entry.eye_catching_image,
        link: entry.link,
        publishedAt: entry.published_at.iso8601,
        title: entry.title,
        feedId: entry.feed_id
      }
    end
  end
end

File: spec/javascript/mocks/Entry.ts
import { Entry } from "@js/types/types";

export const buildEntry = (id: number, feedId: number = id): Entry => {
  return {
    id: id,
    title: `entry title ${id}`,
    description: `entry discription ${id}`,
    link: "https://example.com",
    eyeCatchingImage: "https://example/com/sample.png",
    publishedAt: new Date("2021-05-03T14:35:46+09:00"),
    feedId: feedId,
  };
};
